packages/name-controller/src/NameController.ts

```diff
@@ -612,4 +615,46 @@ export class NameController extends BaseController<
       delete proposedNames[dormantSourceId];
     }
   }
+
+  #removeExpiredEntries(): void {
+    const currentTime = this.#getCurrentTimeSeconds();
+
+    this.update((state: NameControllerState) => {
+      const entries = this.#getEntriesList(state);
+      for (const { nameType, value, variation, entry } of entries) {
+        if (entry.name !== null) {
+          continue;
+        }
+
+        const proposedNames = Object.values(entry.proposedNames);
+        const allProposedNamesExpired = proposedNames.every(
+          (proposedName: ProposedNamesEntry) =>
+            currentTime - (proposedName.lastRequestTime ?? 0) >=
+            PROPOSED_NAME_EXPIRE_DURATION,
+        );
+
+        if (allProposedNamesExpired) {
+          delete state.names[nameType][value][variation];
+        }
+      }
+    });
+  }
+
+  #getEntriesList(state: NameControllerState): {
```

Comment:
Love this method, but do we need the `state` argument?

We have no use for it yet plus the power of this encapsulate is that it transforms all the current state?

Reply:
I had that in mind while creating this method but then got a bit unsure since we use this method in state update callback above. 

Since state update method have the draft state as argument (`this.update( state => {})`) shouldn't we use it there to be sure to get latest? Or do you think it will be safe to use directly from `this.state`?

Reply:
Good point! Definitely best to use the state from the callback as you're doing so we're relying on what the `update` method has given us.